src/cast_core/src/actors/CrudActor.cpp

```diff
@@ -0,0 +1,1241 @@
+            return false;
+        }
+        if (node["BypassDocumentValidation"]) {
+            auto bypassDocValidation = node["BypassDocumentValidation"].as<bool>();
+            rhs.bypass_document_validation(bypassDocValidation);
+        }
+        if (node["Ordered"]) {
+            auto isOrdered = node["Ordered"].as<bool>();
+            rhs.ordered(isOrdered);
+        }
+        if (node["WriteConcern"]) {
+            auto wc = node["WriteConcern"].as<mongocxx::write_concern>();
+            rhs.write_concern(wc);
+        }
+        return true;
+    }
+};
+
+template <>
+struct convert<mongocxx::options::count> {
+    using CountOptions = mongocxx::options::count;
+    static Node encode(const CountOptions& rhs) {
+        Node node;
+        auto h = rhs.hint();
+        if (h) {
+            auto hint = h->to_value().get_utf8().value;
+            node["Hint"] = std::string(hint);
+        }
+        auto limit = rhs.limit();
+        if (limit) {
+            node["Limit"] = *limit;
+        }
+        auto maxTime = rhs.max_time();
+        if (maxTime) {
+            node["MaxTimeMillis"] = genny::TimeSpec(*maxTime);
+        }
+        return node;
+    }
+
+    static bool decode(const Node& node, CountOptions& rhs) {
+        if (!node.IsMap()) {
+            return false;
+        }
+        if (node["Hint"]) {
+            auto h = node["Hint"].as<std::string>();
+            auto hint = mongocxx::hint(h);
+            rhs.hint(hint);
+        }
+        if (node["Limit"]) {
+            auto limit = node["Limit"].as<int>();
+            rhs.limit(limit);
+        }
+        if (node["MaxTimeMillis"]) {
+            auto maxTime = node["MaxTimeMillis"].as<genny::TimeSpec>();
+            rhs.max_time(std::chrono::milliseconds{maxTime});
+        }
+        if (node["ReadPreference"]) {
+            auto readPref = node["ReadPreference"].as<mongocxx::read_preference>();
+            rhs.read_preference(readPref);
+        }
+        return true;
+    }
+};
+
+template <>
+struct convert<mongocxx::options::transaction> {
+    using TransactionOptions = mongocxx::options::transaction;
+    static Node encode(const TransactionOptions& rhs) {
+        Node node;
+        return node;
+    }
+
+    static bool decode(const Node& node, TransactionOptions& rhs) {
+        if (!node.IsMap()) {
+            return false;
+        }
+        if (node["WriteConcern"]) {
+            auto wc = node["WriteConcern"].as<mongocxx::write_concern>();
+            rhs.write_concern(wc);
+        }
+        if (node["ReadConcern"]) {
+            auto rc = node["ReadConcern"].as<mongocxx::read_concern>();
+            rhs.read_concern(rc);
+        }
+        if (node["ReadPreference"]) {
+            auto rp = node["ReadPreference"].as<mongocxx::read_preference>();
+            rhs.read_preference(rp);
+        }
+        return true;
+    }
+};
+}  // namespace YAML
+
+namespace genny::actor {
+
+namespace {
+
+struct BaseOperation {
```

Comment:
nit but I think all these helper things can be just inside an anonymous namespace not inside `genny::actor` - I think just move the opening `namespace genny::actor {` line below all the helper things.